src/Listener/RelatedModelsListener.php

```diff
@@ -87,7 +87,7 @@ public function publishRelatedModels($action = null, $entity = null)
         foreach ($models as $name => $association) {
             list(, $associationName) = pluginSplit($association->name());
             $viewVar = Inflector::variable($associationName);
-            if (array_key_exists($viewVar, $controller->viewVars)) {
+            if (array_key_exists($viewVar, (array)$controller->viewVars)) {
```

Comment:
What if (which it does) IDEs show it as yellow/warning?
Then you would wonder if that code actually works or is correct? Everyone would actually focus on that, which doesn't even have to be in the first place.

Right now it is all good and clearly communicated that this is an array afterwards.